agenta-backend/agenta_backend/routers/observability_router.py

```diff
@@ -0,0 +1,206 @@
+import os
+from typing import List
+
+from fastapi import APIRouter, Depends
+
+from agenta_backend.services.event_db_manager import (
+    costs_of_llm_run,
+    tokens_of_llm_run,
+    latency_of_llm_run,
+    inputs_of_llm_run,
+    outputs_of_llm_run,
+    get_variant_traces,
+    create_app_trace,
+    get_single_trace,
+    update_trace_status,
+    get_trace_spans,
+    add_feedback_to_trace,
+    get_trace_feedbacks,
+    get_feedback_detail,
+    update_trace_feedback,
+)
+from agenta_backend.models.api.observability_models import (
+    Span,
+    CreateFeedback,
+    Feedback,
+    UpdateFeedback,
+    Trace,
+    CreateTrace,
+    UpdateTrace,
+    TraceInputs,
+    TraceOutputs,
+)
+
+if os.environ["FEATURE_FLAG"] in ["cloud", "ee", "demo"]:
+    from agenta_backend.ee.services.auth_helper import (
+        SessionContainer,
+        verify_session,
+    )
+    from agenta_backend.ee.services.selectors import get_user_and_org_id
+else:
+    from agenta_backend.services.auth_helper import (
+        SessionContainer,
+        verify_session,
+    )
+    from agenta_backend.services.selectors import get_user_and_org_id
+
+
+router = APIRouter()
+
+
+@router.post("/traces/", response_model=Trace)
+async def create_trace(
+    payload: CreateTrace,
+    stoken_session: SessionContainer = Depends(verify_session()),
+):
+    # Get user and org id
+    kwargs: dict = await get_user_and_org_id(stoken_session)
+    trace = await create_app_trace(payload, **kwargs)
+    return trace
+
+
+@router.get("/traces/{app_name}/{variant_name}/", response_model=List[Trace])
+async def get_traces(
+    app_name: str,
+    variant_name: str,
+    stoken_session: SessionContainer = Depends(verify_session()),
+):
+    # Get user and org id
+    kwargs: dict = await get_user_and_org_id(stoken_session)
+    traces = await get_variant_traces(app_name, variant_name, **kwargs)
+    return traces
+
+
+@router.get("/traces/{trace_id}/", response_model=Trace)
+async def get_trace(
+    trace_id: str,
+    stoken_session: SessionContainer = Depends(verify_session()),
+):
+    # Get user and org id
+    kwargs: dict = await get_user_and_org_id(stoken_session)
+    trace = await get_single_trace(trace_id, **kwargs)
+    return trace
+
+
+@router.get("/spans/{trace_id}/", response_model=List[Span])
```

Comment:
I don't see post span?
The way I see the api being used in the llm app is that you first post all the spans as they run, and then at the end you post the trace with the ids of the spans. Am I wrong?